photon-lib/src/main/scala/com/linkedin/photon/ml/hyperparameter/estimators/kernels/Matern52.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linkedin.photon.ml.hyperparameter.estimators.kernels
+
+import breeze.linalg.{DenseMatrix, DenseVector}
+import breeze.numerics.{exp, sqrt}
+
+/**
+ * Implements the Matérn 5/2 covariance kernel.
+ *
+ * The Matern kernel is a generalization of the RBF kernel with an additional parameter $\nu$ that allows controlling
+ * smoothness. At $\nu = \infty$, the Matern kernel is equivalent to RBF. At $\nu = 0.5$, it's equivalent to the
+ * absolute exponential kernel. It's noted in the literature that $\nu = 2.5$ allows the kernel to closely approximate
+ * hyperparameter spaces where the smoothness of RBF causes issues (see PBO). Here we hard-code to the 5/2 value because
+ * the computation is much simpler than allowing a user-defined $\nu$.
+ *
+ *     $K(x,x') = \big(\sqrt{5r^2(x,x')} + \frac{5}{3} r^2(x,x') + 1\big) \exp(-\sqrt{5r^2(x,x')})$
```

Comment:
It's hard-coded to 5/2 to make the implementation simpler. If we allow specifying `nu`, the computation becomes much more complex.